services/src/datasets/external/natur40.rs

```diff
@@ -0,0 +1,290 @@
+use std::path::Path;
+
+use crate::error::Error;
+use crate::util::parsing::string_or_string_array;
+use crate::{datasets::listing::DatasetListOptions, error::Result};
+use crate::{
+    datasets::{
+        listing::{DatasetListing, DatasetProvider},
+        storage::DatasetProviderDefinition,
+    },
+    error,
+    util::user_input::Validated,
+};
+use async_trait::async_trait;
+use futures::future::join_all;
+use gdal::DatasetOptions;
+use geoengine_datatypes::dataset::{DatasetId, DatasetProviderId, ExternalDatasetId};
+use geoengine_operators::engine::TypedResultDescriptor;
+use geoengine_operators::source::GdalMetaDataStatic;
+use geoengine_operators::util::gdal::{
+    gdal_open_dataset_ex, gdal_parameters_from_dataset, raster_descriptor_from_dataset,
+};
+use geoengine_operators::{
+    engine::{
+        MetaData, MetaDataProvider, RasterQueryRectangle, RasterResultDescriptor,
+        VectorQueryRectangle, VectorResultDescriptor,
+    },
+    mock::MockDatasetDataSourceLoadingInfo,
+    source::{GdalLoadingInfo, OgrSourceDataset},
+};
+use log::info;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Natur40DataProviderDefinition {
+    id: DatasetProviderId,
+    name: String,
+    base_url: String,
+    user: String,
+    password: String,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl DatasetProviderDefinition for Natur40DataProviderDefinition {
+    async fn initialize(self: Box<Self>) -> crate::error::Result<Box<dyn DatasetProvider>> {
+        Ok(Box::new(Natur40DataProvider {
+            id: self.id,
+            base_url: self.base_url,
+            user: self.user,
+            password: self.password,
+        }))
+    }
+
+    fn type_name(&self) -> String {
+        "Natur4.0".to_owned()
```

Comment:
```suggestion
        "Nature 4.0".to_owned()
```

Reply:
currently we don't have spaces in type names. they aren't supposed to be used as text anyway (even though currently they are) but to distinguish different types of providers